src/widget/wbattery.cpp

```diff
@@ -159,10 +161,10 @@ void WBattery::paintEvent(QPaintEvent* /*unused*/) {
     p.drawPrimitive(QStyle::PE_Widget, option);
 
     if (m_pPixmapBack) {
-        m_pPixmapBack->draw(0, 0, &p);
+        m_pPixmapBack->draw(rect(), &p);
     }
 
     if (m_pCurrentPixmap) {
-        m_pCurrentPixmap->draw(0, 0, &p);
+        m_pCurrentPixmap->draw(rect(), &p);
```

Comment:
I don't really understand this change, it seems to be needed to make the scaling work..

Reply:
The original version draws a into a rect taken from the prixamp placed at 0,0 of the widget.
The new version takes the whole widget rect as desired.